### PR TITLE
Add Dockerfile and Docker CI workflow

### PR DESCRIPTION
**Pull request by vladsavelyev**
Currently, if we want to test recent cpg-utils changes, we have to rebuild the driver image. This change allows to run analysis-runner with `--image australia-southeast1-docker.pkg.dev/cpg-common/images/cpg_utils:4.8.12` for faster development cycle. 

It would probably be even handier to automatically build a SHA-tagged image for every pull request as well, to allow a faster development cycle in the test namespace.

### CODE REVIEW

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Docker
+on:
+  push:
+    branches:
+      - main
+
+env:
+  VERSION: 4.8.12
```
Comment (lgruen) [inline .github/workflows/docker.yaml]: Should we use the GitHub SHA as part of the tag instead? Or can we extract the version from one of the files that `bumpversion` updates?

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM australia-southeast1-docker.pkg.dev/analysis-runner/images/driver:62e79e54d43aa1762ec68a6661872320a39517bd-hail-2ea2615a797a5aff72d20d9d12a2609342846a07
```
Comment (lgruen) [inline docker/Dockerfile]: Should we add a `latest` tag to the `driver` image, so we don't need to hardcode a version here (which will immediately become obsolete)?

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+ARG VERSION=4.8.13
+
+RUN python3 -m pip install cpg_utils[workflows]==$VERSION
```
Comment (lgruen) [inline docker/Dockerfile]: Won't this result in a race condition with the package generation + publishing?

Should we instead build the package locally within the Docker container by copying the files from the checked out repo?

**Review by lgruen — CHANGES_REQUESTED** [review]

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (vladsavelyev) [reply .github/workflows/docker.yaml]: `bump2version` updates this value directly

File: docker/Dockerfile (hunk 2 above)
Comment (vladsavelyev) [reply docker/Dockerfile]: Yes, that would be amazing!

File: docker/Dockerfile (hunk 3 above)
Comment (vladsavelyev) [reply docker/Dockerfile]: Good point! I'll change that

File: docker/Dockerfile (hunk 2 above)
Comment (vladsavelyev) [reply docker/Dockerfile]: https://github.com/populationgenomics/analysis-runner/pull/532

File: docker/Dockerfile (hunk 3 above)
Comment (vladsavelyev) [reply docker/Dockerfile]: Done

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FROM australia-southeast1-docker.pkg.dev/analysis-runner/images/driver:62e79e54d43aa1762ec68a6661872320a39517bd-hail-2ea2615a797a5aff72d20d9d12a2609342846a07
```
Comment (lgruen) [inline Dockerfile]:
```suggestion
FROM australia-southeast1-docker.pkg.dev/analysis-runner/images/driver:latest
```

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+ADD README.md .
+ADD cpg_utils .
+ADD setup.py .
```
Comment (lgruen) [inline Dockerfile]: `COPY` is generally preferable to `ADD`: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy

```suggestion
COPY README.md .
COPY cpg_utils .
COPY setup.py .
```

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (lgruen) [reply .github/workflows/docker.yaml]: Oh, how nice, I completely missed the `[bumpversion:file:.github/workflows/docker.yaml]` part!

**Review by lgruen — APPROVED** [review]

**Pull request merged**